From 5e5cf6fc59db2322dfe3ac8e1002f066b06d868f Mon Sep 17 00:00:00 2001
From: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Date: Thu, 6 May 2010 13:40:18 -0700
Subject: USB: xhci: Set stream ID to 0 after cleaning up stalls.
Patch-mainline: v2.6.35
Git-commit: 5e5cf6fc59db2322dfe3ac8e1002f066b06d868f

After using state stored in xhci_virt_ep to clean up a stalled endpoint,
be sure to set the stalled stream ID back to 0.

Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/xhci-ring.c |    1 +
 drivers/usb/host/xhci.c      |    1 +
 2 files changed, 2 insertions(+)

--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -680,6 +680,7 @@ remove_finished_td:
 	}
 	ep->stopped_td = NULL;
 	ep->stopped_trb = NULL;
+	ep->stopped_stream = 0;
 
 	/*
 	 * Drop the lock and complete the URBs in the cancelled TD list.
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -1492,6 +1492,7 @@ void xhci_endpoint_reset(struct usb_hcd
 	}
 	virt_ep->stopped_td = NULL;
 	virt_ep->stopped_trb = NULL;
+	virt_ep->stopped_stream = 0;
 	spin_unlock_irqrestore(&xhci->lock, flags);
 
 	if (ret)
